### PR TITLE
EN-908 Asynchronous CryptKeyAccessor callback

### PR DESCRIPTION
**Pull request by vladimir-grezin**
These changes are required to accomplish https://incountry.atlassian.net/browse/EN-908
`CryptKeyAccessor` now supports both sync and async `getKeySecurely` callback.

There is an `FIXME` at `crypt-key-accessor.js:11`. I'm not sure how we manage versions.

### CODE REVIEW

File: node-sdk/crypt-key-accessor.js [hunk 1]
```diff
@@ -5,19 +5,25 @@ class CryptKeyAccessor {
...
+     * @return Promise which resolve to key string
      */
-    secureAccessor(useKeySecurelyCallback) {
-        let key = {
-            "closedOver": this._getKeySecurely()
-        };
-
-        useKeySecurelyCallback(key.closedOver);
-        delete key.closedOver;
+    secureAccessor() {
```
Comment (joelbruce-ic) [inline node-sdk/crypt-key-accessor.js]: Great catch, Love it!

File: node-sdk/in-crypt.js [hunk 2]
```diff
@@ -29,61 +29,49 @@ class InCrypt {
...
-                    let paddedBytes = aesjs.utils.utf8.toBytes(padded);
-                    let encryptedBytes = aesCbc.encrypt(paddedBytes);
-                    let encryptedHex = aesjs.utils.hex.fromBytes(encryptedBytes);
-                    resolve(encryptedHex);
-               });
-            }
-            catch (err) {
-                reject(err)
-            }
+        return new Promise((resolve, reject) => {
```
Comment (joelbruce-ic) [inline node-sdk/in-crypt.js]: Another great catch!

File: node-sdk/tests/unit/in-crypt-tests.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
-var {InCrypt, pad, unpad} = require('../../in-crypt');
-var CryptKeyAccessor = require('../../crypt-key-accessor');
+const {InCrypt, pad, unpad} = require('../../in-crypt');
```
Comment (joelbruce-ic) [inline node-sdk/tests/unit/in-crypt-tests.js]: I am enjoying seeing that you know best practices, and I appreciate you cleaning up the older code. Really good stuff!

File: node-sdk/tests/unit/in-crypt-tests.js [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
+const CryptKeyAccessor = require('../../crypt-key-accessor');
 
-var expect = require('chai').expect;
+const expect = require('chai').expect;
 
 describe('InCrypt', function() {
-    context('with variable length unencrypted text', function() {
+    context('with constiable length unencrypted text', function() {
```
Comment (joelbruce-ic) [inline node-sdk/tests/unit/in-crypt-tests.js]: What does constiable mean? Did you mean constable? Is that a Russian word? :)

File: node-sdk/tests/unit/in-crypt-tests.js [hunk 5]
```diff
@@ -26,21 +26,36 @@ describe('InCrypt', function() {
...
                 console.log(`e: ${encrypted}`);
 
-                var decrypted = await incrypt.decryptAsync(encrypted);
+                const decrypted = await incrypt.decryptAsync(encrypted);
+                console.log(`d: ${decrypted}`);
+                expect(decrypted).to.equal(testCase);
+                expect(encrypted).to.not.equal(decrypted);
+            })
+
+            it(`should encrypt and decrypt correctly (synchronous accessor): ${testCase}`, async function() {
```
Comment (joelbruce-ic) [inline node-sdk/tests/unit/in-crypt-tests.js]: I like that you made synchronous and asynchronous test cases.

File: node-sdk/tests/unit/in-crypt-tests.js (hunk 4 above)
Comment (vladimir-grezin) [reply node-sdk/tests/unit/in-crypt-tests.js]: Ah, lol. This it's autocorrect error. My fault

**Pull request merged**